src/dotnet/ToolPackage/IToolPackageStore.cs

```diff
@@ -4,13 +4,26 @@
 using System;
 using System.Collections.Generic;
 using Microsoft.Extensions.EnvironmentAbstractions;
+using NuGet.Versioning;
 
 namespace Microsoft.DotNet.ToolPackage
 {
     internal interface IToolPackageStore
     {
         DirectoryPath Root { get; }
 
-        IEnumerable<IToolPackage> GetInstalledPackages(string packageId = null);
+        DirectoryPath GetRandomStagingDirectory();
+
+        NuGetVersion GetStagedPackageVersion(DirectoryPath stagingDirectory, PackageId packageId);
+
+        DirectoryPath GetRootPackageDirectory(PackageId packageId);
+
+        DirectoryPath GetPackageDirectory(PackageId packageId, NuGetVersion version);
+
+        IEnumerable<IToolPackage> EnumeratePackages();
```

Comment:
nit: i think Packages is enough, the return type is IEnumerable, Enumerable is redundant

Reply:
My rationale for using `Enumerate` as part of the method name: 1) There's precedent in corefx (e.g. `Directory.EnumerateFileSystemEntries`) and 2) given the name, you know it is iterating rather than returning a complete enumerable collection.

I'm fine with foregoing `Enumerate` with another verb, but a verb `Get` makes it sound like it's returning all of the packages at once.